custom_typings/estree.d.ts

```diff
@@ -0,0 +1,411 @@
+declare module 'estree' {
```

Comment:
I swear I looked for such a thing, I don't know how I missed this.

Annoyingly, now that I've written my own I can see some ways in which theirs are worse.

Added a TODO to merge and rationalize.